geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureConverterFactory.scala

```diff
@@ -234,8 +315,16 @@ trait ToSimpleFeatureConverter[I] extends SimpleFeatureConverter[I] with LazyLog
     new EvaluationContextImpl(names, values, counter)
   }
 
-  override def processInput(is: Iterator[I], ec: EvaluationContext): Iterator[SimpleFeature] =
-    is.flatMap(i =>  processSingleInput(i, ec))
+  override def processInput(is: Iterator[I], ec: EvaluationContext): Iterator[SimpleFeature] = {
+    parseOpts.parseMode match {
+       case ParseMode.Incremental  =>
+         is.flatMap(i => processSingleInput(i, ec))
+       case ParseMode.Batch =>
```

Comment:
hmmm. that's interesting. perhaps this is better renamed "full file" or maybe we can do batch size of 0 in batch mode means full file? the idea was meant to have it be able to parse the whole file and insert no records. so it can still accomplish that. else you don't need batch